src/omv/imlib/imlib.h

```diff
@@ -336,75 +336,134 @@ extern const uint16_t ironbow_table[256];
 // Image Stuff //
 /////////////////
 
+// Pixel format Byte Per Pixel.
 typedef enum {
-    PIXFORMAT_INVALID = 0,
-    PIXFORMAT_BINARY,    // 1BPP/BINARY
-    PIXFORMAT_GRAYSCALE, // 1BPP/GRAYSCALE
-    PIXFORMAT_RGB565,    // 2BPP/RGB565
-    PIXFORMAT_YUV422,    // 2BPP/YUV422
-    PIXFORMAT_BAYER,     // 1BPP/RAW
-    PIXFORMAT_JPEG,      // JPEG/COMPRESSED
+    PIXFORMAT_BPP_BINARY    = 0,
+    PIXFORMAT_BPP_GRAYSCALE = 1,
+    PIXFORMAT_BPP_RGB565    = 2,
+    PIXFORMAT_BPP_YUV422    = 2,
+    PIXFORMAT_BPP_BAYER     = 1,
+} pixformat_bpp_t;
+
+// Pixel format IDs.
+typedef enum {
+  PIXFORMAT_ID_BAYER_BGGR   = 0, // < Note: about bayer sub-formats >
```

Comment:
I may add more bayer formats to support automotive sensors that have clear filters...

Reply:
Wouldn't that be grayscale ? Anyway so I switched back to the original format split:

```c
// 31......28  27  26  25  24  23..........16  15...........8  7.............0
// <RESERVED>  MF  CF  JF  RF  <PIXFORMAT_ID>  <SUBFORMAT_ID>  <BYTES_PER_PIX>
// NOTE: Bit 31-30 must Not be used for pixformat_t to be used as mp_int_t.
```
And increased `sensor.hw_flags.bayer` to 3 bits (8 possible sub-formats), you can add more if needed. You can also test if an image is *any* bayer by using `image->pixfmt_id == PIXFORMAT_ID_BAYER` and you can either switch on `img->pixfmt`, `img->pixfmt_id` or `img->subfmt_id` for subformats.